applications/StructuralMechanicsApplication/python_scripts/symbolic_generation/total_lagrangian_mixed_volumetric_strain_element/total_lagrangian_mixed_volumetric_strain_element_template.cpp

```diff
@@ -0,0 +1,1261 @@
+    auto& r_cons_law_options = cons_law_values.GetOptions();
+    r_cons_law_options.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
+    r_cons_law_options.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
+    r_cons_law_options.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true);
+
+    // Calculate the RHS and LHS contributions
+    rLeftHandSideMatrix.clear();
+    rRightHandSideVector.clear();
+
+    // Calculate stabilization constant
+    const double c_tau_u = 2.0;
+    const double c_tau_th = 0.1;
+    const double h = ElementSizeCalculator<2,NumNodes>::AverageElementSize(r_geometry);
+    const double aux_tau = c_tau_u * std::pow(h,2) / 2.0;
+
+    // Set the auxiliary references matching the automatic differentiation symbols
+    array_1d<double,3> b_gauss;
+    const auto& N = kinematic_variables.N;
+    const auto& DN = kinematic_variables.DN_DX;
+    const auto& u = kinematic_variables.Displacements;
+    const auto& th = kinematic_variables.JacobianDeterminant;
+    const auto& S = constitutive_variables.StressVector;
+    const auto& C = constitutive_variables.ConstitutiveMatrix;
+
+    // RHS and LHS assembly
+    const SizeType n_gauss = r_geometry.IntegrationPointsNumber(GetIntegrationMethod());
+    const auto& r_integration_points = r_geometry.IntegrationPoints(GetIntegrationMethod());
+    for (IndexType i_gauss = 0; i_gauss < n_gauss; ++i_gauss) {
+        // Calculate kinematics
+        CalculateKinematicVariables(kinematic_variables, i_gauss, GetIntegrationMethod());
+        const double w_g = kinematic_variables.detJ0 * r_integration_points[i_gauss].Weight();
+
+        // Calculate the constitutive response
+        CalculateConstitutiveVariables(
+            kinematic_variables,
+            constitutive_variables,
+            cons_law_values,
+            i_gauss,
+            r_geometry.IntegrationPoints(this->GetIntegrationMethod()),
+            ConstitutiveLaw::StressMeasure_PK2);
+
+        // Calculate body force
+        // Note that this already includes the density computed in the reference configuration
+        b_gauss = StructuralMechanicsElementUtilities::GetBodyForce(*this, r_integration_points, i_gauss);
+
+        // Calculate the stabilization constants
+        const double kappa = CalculateBulkModulus(constitutive_variables.ConstitutiveMatrix); // Equivalent bulk modulus
+        const double mu = std::min(CalculateShearModulus(constitutive_variables.ConstitutiveMatrix), mMinShearModulusVector[i_gauss]); // 2nd Lame constant (shear modulus)
+        mMinShearModulusVector[i_gauss] = mu; // Save the current 2nd Lame constant for the next iteration checks (note that we use the minimum historical value)
+        const double tau_u = aux_tau / mu;
+        const double tau_th = (c_tau_th * mu) / (mu + kappa);
+
+        // Calculate and add the LHS Gauss point contributions
+        //substitute_lhs_2D_3N
```

Comment:
is this needed?

Reply:
Yes. This is the template to be filled by the automatic differentiation tool. Once the differentiation is completed, it parses this file and search for this particular string to be substituted.